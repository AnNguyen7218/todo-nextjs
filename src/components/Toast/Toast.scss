@import 'config.gutters';
@import 'config.media';
@import 'config.variables';
@import 'config.colors';

$width-sm: 85vw;
$width-lg: 34rem;

.toast-container {
  width: $width-sm;
  left: 50%;
  top: gutter(8);
  margin-left: calc(#{$width-sm} / -2);

  @include break(lg) {
    width: $width-lg;
    margin-left: calc(#{$width-lg} / -2);
  }
}

.toast {
  padding: gutter(8) gutter(16);
  min-height: auto;
  max-height: unset;
  width: auto;
  box-shadow: $elevator-1;
  border: 0.1rem solid;
  margin-bottom: gutter(8);

  &--success {
    background-color: $toast-success-bg;
    border-color: $toast-success-border;
  }

  &--error {
    background-color: $toast-error-bg;
    border-color: $toast-error-border;
  }

  &--warning {
    background-color: $toast-warning-bg;
    border-color: $toast-warning-border;
  }
}

.toast-body {
  margin: gutter(0);
}
