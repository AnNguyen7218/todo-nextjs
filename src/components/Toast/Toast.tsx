import React from 'react';
import cx from 'classnames';
import { ToastContainer, Slide } from 'react-toastify';
import 'react-toastify/dist/ReactToastify.css';
import css from './Toast.scss';

type Props = {
  theme: ToastTheme;
};

const Toast: React.FC<Props> = ({ theme }) => {
  return (
    <ToastContainer
      className={css['toast-container']}
      toastClassName={cx(css['toast'], css[`toast--${theme}`])}
      bodyClassName={css['toast-body']}
      hideProgressBar
      closeButton={false}
      draggable={false}
      transition={Slide}
      pauseOnFocusLoss={false}
    />
  );
};

export default Toast;
