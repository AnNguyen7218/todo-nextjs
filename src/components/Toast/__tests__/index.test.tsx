import React from 'react';
import { render, cleanup, fireEvent } from '@testing-library/react';
import { toast } from 'react-toastify';
import Toast from '../Toast';

afterEach(() => {
  cleanup();
  jest.useFakeTimers();
});

const toastMessage = (message: string) => toast(message);

describe('Toast Component', () => {
  test('Toast should match snapshot', () => {
    const { container } = render(<Toast theme="success" />);
    expect(container).toMatchSnapshot();
  });

  describe('Toast should appear', () => {
    test('Toast notify as success', () => {
      const { getByText, asFragment } = render(
        <>
          <button
            onClick={() => {
              toastMessage('Success');
            }}
          >
            Toast
          </button>
          <Toast theme="success" />
        </>
      );
      const button = getByText('Toast');
      fireEvent.click(button);
      jest.runAllTimers();
      expect(asFragment()).toMatchSnapshot();
    });

    test('Toast notify as error', () => {
      const { getByText, asFragment } = render(
        <>
          <button
            onClick={() => {
              toastMessage('Error');
            }}
          >
            Toast
          </button>
          <Toast theme="error" />
        </>
      );
      const button = getByText('Toast');
      fireEvent.click(button);
      jest.runAllTimers();
      expect(asFragment()).toMatchSnapshot();
    });

    test('Toast notify as warning', () => {
      const { getByText, asFragment } = render(
        <>
          <button
            onClick={() => {
              toastMessage('Warning');
            }}
          >
            Toast
          </button>
          <Toast theme="warning" />
        </>
      );
      const button = getByText('Toast');
      fireEvent.click(button);
      jest.runAllTimers();
      expect(asFragment()).toMatchSnapshot();
    });
  });
});
