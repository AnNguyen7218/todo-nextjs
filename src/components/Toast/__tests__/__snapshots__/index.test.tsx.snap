// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toast Component Toast should appear Toast notify as error 1`] = `
<DocumentFragment>
  <button>
    Toast
  </button>
  <div
    class="Toastify"
  >
    <div
      class="Toastify__toast-container Toastify__toast-container--top-right toast-container"
    >
      <div
        class="Toastify__toast Toastify__toast--default toast toast--error"
        style=""
      >
        <div
          class="Toastify__toast-body toast-body"
          role="alert"
        >
          Error
        </div>
        <div
          class="Toastify__progress-bar Toastify__progress-bar--animated Toastify__progress-bar--default"
          style="animation-duration: 5000ms; animation-play-state: running; opacity: 0;"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Toast Component Toast should appear Toast notify as success 1`] = `
<DocumentFragment>
  <button>
    Toast
  </button>
  <div
    class="Toastify"
  >
    <div
      class="Toastify__toast-container Toastify__toast-container--top-right toast-container"
    >
      <div
        class="Toastify__toast Toastify__toast--default toast toast--success"
        style=""
      >
        <div
          class="Toastify__toast-body toast-body"
          role="alert"
        >
          Success
        </div>
        <div
          class="Toastify__progress-bar Toastify__progress-bar--animated Toastify__progress-bar--default"
          style="animation-duration: 5000ms; animation-play-state: running; opacity: 0;"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Toast Component Toast should appear Toast notify as warning 1`] = `
<DocumentFragment>
  <button>
    Toast
  </button>
  <div
    class="Toastify"
  >
    <div
      class="Toastify__toast-container Toastify__toast-container--top-right toast-container"
    >
      <div
        class="Toastify__toast Toastify__toast--default toast toast--warning"
        style=""
      >
        <div
          class="Toastify__toast-body toast-body"
          role="alert"
        >
          Warning
        </div>
        <div
          class="Toastify__progress-bar Toastify__progress-bar--animated Toastify__progress-bar--default"
          style="animation-duration: 5000ms; animation-play-state: running; opacity: 0;"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Toast Component Toast should match snapshot 1`] = `
<div>
  <div
    class="Toastify"
  />
</div>
`;
